pkg/kubelet/cm/dra/manager.go

```diff
@@ -124,6 +124,13 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 		// in the cache.
 		claimInfo.addPodReference(pod.UID)
 
+		// Sync to Checkpoint in advance to prevent the loss of claim requests.
```

Comment:
This is somehow contradicts with the comment above (lines 117-123). Either that comment is incorrect and we should modify or delete it, or there is no need in caching and checkpointing here.

Reply:
@ruquanzhao please address above comment and add e2e test to cover the issue, thanks